syshook_utils.c

```diff
@@ -0,0 +1,64 @@
+struct sys_hook {
+    void** orig_func_pointer;
+    void* hook_func;
+    int sys_entry;
+};
+
+int set_page_rw(unsigned long addr)
+{
+    unsigned int level;
+    pte_t *pte = lookup_address(addr, &level);
+    if (pte->pte &~ _PAGE_RW) pte->pte |= _PAGE_RW;
+    return 0;
+}
+
+int set_page_ro(unsigned long addr)
+{
+    unsigned int level;
+    pte_t *pte = lookup_address(addr, &level);
+    pte->pte = pte->pte &~_PAGE_RW;
+    return 0;
+}
+
+int set_sys_hooks(struct sys_hook *hooks){
+    int res;
+    int i;
+    unsigned long **p_sys_call_table = (void *) kallsyms_lookup_name("sys_call_table");
+
+    if(!p_sys_call_table){
+        printk("failed to get sys_call_table addres");
+        return -1;
+    }
+
+    res = set_page_rw((unsigned long)p_sys_call_table);
+    if (res != 0){
+        printk("failed to set write permissions for systable");
+        return -1;
+    }
+
+    for (i = 0; i < 1; i++) {
```

Comment:
if your condition is `i < 1` you don't need a loop.

Support more than 1 hook

Reply:
done